src/esp/scene/SceneNode.cpp

```diff
@@ -16,16 +18,61 @@ SceneNode::SceneNode(SceneNode& parent)
   setId(parent.getId());
   setCachedTransformations(Mn::SceneGraph::CachedTransformation::Absolute);
   absoluteTransformation_ = absoluteTransformation();
+  nodeSensorSuite_ = new esp::sensor::SensorSuite(*this);
+  subtreeSensorSuite_ = new esp::sensor::SensorSuite(*this);
 }
 
 SceneNode::SceneNode(MagnumScene& parentNode)
     : Mn::SceneGraph::AbstractFeature3D{*this} {
   setParent(&parentNode);
   setCachedTransformations(Mn::SceneGraph::CachedTransformation::Absolute);
   absoluteTransformation_ = absoluteTransformation();
+  nodeSensorSuite_ = new esp::sensor::SensorSuite(*this);
+  subtreeSensorSuite_ = new esp::sensor::SensorSuite(*this);
+}
+
+SceneNode::~SceneNode() {
+  // if the entire scene graph is being deleted, no need to update anything
+  if (SceneGraph::isRootNode(*this)) {
+    return;
+  }
+  // if it is to delete a subtree
+  // then two cases:
+  // 1) current node is the root node (of this subtree).
+  // In case 1, you do not have to do anything since its parent node is nullptr
+  // already!! This is because magnum destroys the nodes in a top-down manner
+  // recursively, root node first, and then its child nodes
+  if (!this->parent()) {
+    return;
+  }
+
+  // 2) current node is NOT the root node.
+  // In case 2, you need update the sensorSuites stored in each ancestor node in
+  // a bottom-up manner.
+
+  // First erase all nodes in nodeSensorSuite from its parent's nodeSensorSuite
```

Comment:
@vauduong :  I thought about it. The following for loop is actually only needed for the LEAF nodes. Is it correct?

If yes, we can add a if clause.
If no, let me know your thoughts.

Reply:
Yes, I think this is true. 
If it's a leaf node, that means its parent node will have its sensor in the nodeSensorSuite.
If it's not a leaf node, then that means any sensors in its nodeSensorSuite are child Sensors, and will just be deleted when the node is deleted.

Reply:
If node "A" is not a leaf node, any sensor in the `nodeSensorSuite_` of "A" will not be in the nodeSensorSuite_ of "A"'s parent for sure.